QGIS_plugin/bgtinlooptool/scripts/2_test_write_to_gpkg.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Aug  5 09:40:20 2024
+
+@author: ruben.vanderzaag
+"""
+
+from osgeo import ogr
+
+def _write_to_disk(file_path, db_layer_name, dst_layer_name): 
+    """Copy mem_database to file_path"""
+    # Get the source layer from the memory database
+    db_layer = mem_database.GetLayerByName(db_layer_name)
+    if db_layer is None:
+        raise ValueError(f"Layer '{db_layer_name}' not found in memory database.")
+    
+    # Open the destination GeoPackage in write mode
+    dst_gpkg = GPKG_DRIVER.Open(file_path, 1)  # 1 means writable
```

Comment:
Het lijkt me beter om dit met een context manager te doen. Er staat nu wel een dst_gpkg = None aan het eind, maar nu is er wel gevaar dat het bestand nog open is als er iets in de code fout gaat.

context manager ziet er vast bekend uit:

 with GPKG_DRIVER.Open(file_path, 1) as dst_gpkg:
     ..hier de code